src/guide/agent-functions/defining-agent-functions.rst

```diff
@@ -74,12 +84,128 @@ When using the Run-Time Compiled (RTC) functions, optionally in the C++ API or r
+``numpy.intc``         A numpy ``intc`` type which will transpile to a C++ ``int``
+``numpy.uintc``        A numpy ``uintc`` type which will transpile to a C++ ``unsigned int``
+``numpy.uint``         A numpy ``uint`` type which will transpile to a C++ ``unsigned int``
+``numpy.longlong``     A numpy ``longlong`` type which will transpile to a C++ ``long long``
+``numpy.ulonglong``    A numpy ``ulonglong`` type which will transpile to a C++ ``unsigned long long``
+``numpy.half``         A numpy ``half`` type which will transpile to a C++ ``half``
+``numpy.single``       A numpy ``single`` type which will transpile to a C++ ``float``
+``numpy.double``       A numpy ``double`` type which will transpile to a C++ ``double``
+``numpy.longdouble``   A numpy ``longdouble`` type which will transpile to a C++ ``long double``
+``numpy.bool_``        A numpy ``bool_`` type which will transpile to a C++ ``bool``
+``numpy.bool8``        A numpy ``bool8`` type which will transpile to a C++ ``bool``
+``numpy.int_``         A numpy ``int_`` type which will transpile to a C++ ``long``
+``numpy.int8``         A numpy ``int8`` type which will transpile to a C++ ``int8_t``
+``numpy.int16``        A numpy ``int16`` type which will transpile to a C++ ``int16_t``
+``numpy.int32``        A numpy ``int32`` type which will transpile to a C++ ``int32_t``
+``numpy.int64``        A numpy ``int64`` type which will transpile to a C++ ``int64_t``
+``numpy.intp``         A numpy ``intp`` type which will transpile to a C++ ``intptr_t``
+``numpy.uint_``        A numpy ``uint_`` type which will transpile to a C++ ``long``
+``numpy.uint8``        A numpy ``uint8`` type which will transpile to a C++ ``uint8_t``
+``numpy.uint16``       A numpy ``uint16`` type which will transpile to a C++ ``uint16_t``
+``numpy.uint32``       A numpy ``uint32`` type which will transpile to a C++ ``uint32_t``
+``numpy.uint64``       A numpy ``uint64`` type which will transpile to a C++ ``uint64_t``
+``numpy.uintp``        A numpy ``uintp`` type which will transpile to a C++ ``uintptr_t``
+``numpy.float_``       A numpy ``float_`` type which will transpile to a C++ ``float``
+``numpy.float16``      A numpy ``float16`` type which will transpile to a C++ ``half``
+``numpy.float32``      A numpy ``float32`` type which will transpile to a C++ ``float``
+``numpy.float64``      A numpy ``float64`` type which will transpile to a C++ ``double``
+====================== =================================================================================
+
+Any runtime compiled agent functions must include the definition of any device functions in the agent function string. These can not be shared between agent functions (as each has a unique string definition).
+
+Pure python device functions must be contained in the source module of any agent functions which call them. Device functions are automatically discovered and included in the transpilation process for agent agent functions in the same Python module. 
```

Comment:
Perhaps clarify that module means file in this context? Or does it just mean has to be imported by the file?

Reply:
Have clarified this